Add some image and code examples about TextBox (#59)
AvaloniaUI/avaloniaui.net: wwwroot/docs/controls/textbox.md

Title: Textbox
---

> The `TextBox` is editable text field where user can input text.

Reviewer: ```suggestion
The `TextBox` control is an editable text field where a user can input text.
```

Author: Changed in second commit.

# Source code
[TextBox.cs](https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Controls/TextBox.cs)

# Examples

## Basic one line TextBox

```xml
<Window xmlns="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        mc:Ignorable="d" d:DesignWidth="800" d:DesignHeight="450"
        x:Class="AvaloniaAppTemplate.MainWindow"
        Title="AvaloniaAppTemplate">
	<StackPanel Margin="10">
		<TextBox />
	</StackPanel>
</Window>
```

> produces following output with **Windows 10**  

Reviewer: ```suggestion
produces the following output in **Windows 10**  
```

Author: Changed in second commit.

![Basic TextBox](images/textbox_basic.png)

## Password input TextBox

```xml
<Window xmlns="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        mc:Ignorable="d" d:DesignWidth="800" d:DesignHeight="450"
        x:Class="AvaloniaAppTemplate.MainWindow"
        Title="AvaloniaAppTemplate">
	<StackPanel Margin="10">
		<TextBox PasswordChar="#" />
	</StackPanel>
</Window>
```

> produces following output with **Windows 10** when text is type to input  

Reviewer: ```suggestion
produces the following output with **Windows 10** when text is input  
```

Author: Changed in second commit.

![Password TextBox](images/textbox_password.png)

## TextBox with watermark

> With Avalonia and WPF text suggestion that textbox shows is called watermark, but e.g. HTML5 calls it *placeholder*

Reviewer: This is a little hard to understand, and what's more I don't think WPF has this feature. Might something like

```suggestion
Avalonia can show a "watermark" in a `TextBox`, which is a piece of text that is displayed when the `TextBox` is empty (in HTML5 this is called a *placeholder*)
```

be better?

Author: Yeah, sorry I confused plain WPF to MahApps WPF. I used your suggested text in second commit.

```xml
<Window xmlns="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        mc:Ignorable="d" d:DesignWidth="800" d:DesignHeight="450"
        x:Class="AvaloniaAppTemplate.MainWindow"
        Title="AvaloniaAppTemplate">
	<StackPanel Margin="10">
		<TextBox Watermark="Street address" />
	</StackPanel>
</Window>
```

> produces following output with **Windows 10**  

Reviewer: ```suggestion
produces the following output with **Windows 10**  
```

Author: Changed in second commit.

![Password TextBox](images/textbox_watermark.png)

## Multiline TextBox

```xml
<Window xmlns="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        mc:Ignorable="d" d:DesignWidth="800" d:DesignHeight="450"
        x:Class="AvaloniaAppTemplate.MainWindow"
        Title="AvaloniaAppTemplate">
	<Grid Margin="10">
		<TextBox AcceptsReturn="True" TextWrapping="Wrap" />
	</Grid>
</Window>
```

> produces following output with **Windows 10** when text is type to input  

Reviewer: ```suggestion
produces the following output with **Windows 10** when text is input  
```

Author: Changed in second commit.
